src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -119,93 +118,82 @@ public MergeOrchestratorService(NetworkStoreService networkStoreService,
 
-            if (checkTso(processConfigs, tso, format, businessProcess)) {
-                // required tso received
-                ZonedDateTime dateTime = ZonedDateTime.parse(date);
+            // Get all process configs concerned
+            List<ProcessConfig> processConfigsConcerned = mergeConfigService.getConfigs().stream()
+                    .filter(c -> c.isMatching(tso, format, businessProcess))
+                    .collect(Collectors.toList());
 
-                for (ProcessConfig processConfig : processConfigs) {
-                    if (processConfig.getTsos().stream().anyMatch(ts -> ts.equals(tso)) &&
-                            processConfig.getBusinessProcess().equals(businessProcess)) {
-                        LOGGER.info("Merge {} of process {} {} : IGM in format {} from TSO {} received", date, processConfig.getProcess(), processConfig.getBusinessProcess(), format, tso);
-                        mergeEventService.addMergeIgmEvent(processConfig.getProcess(), processConfig.getBusinessProcess(), dateTime, tso, IgmStatus.AVAILABLE, null, null, null, null);
-                    }
-                }
+            // Send all availability messages
+            processConfigsConcerned.forEach(processConfig -> {
+                LOGGER.info("Merge {} of process {} {} : IGM in format {} from TSO {} received", date, processConfig.getProcess(), processConfig.getBusinessProcess(), format, tso);
 
-                if (!processConfigs.isEmpty()) {
-                    // import IGM into the network store
-                    UUID networkUuid = caseFetcherService.importCase(caseUuid);
-                    // check IGM quality
-                    boolean valid = igmQualityCheckService.check(networkUuid);
+                // if already received delete old network
+                Optional<IgmEntity> igmEntityOptional = igmRepository.findByProcessAndDateAndTso(processConfig.getProcess(), LocalDateTime.ofInstant(dateTime.toInstant(), ZoneOffset.UTC), tso);
+                igmEntityOptional.ifPresent(igmEntity -> networkStoreService.deleteNetwork(igmEntity.getNetworkUuid()));
 
-                    merge(processConfigs.get(0), dateTime, date, tso, valid, networkUuid, caseUuid, businessProcess, null, null);
+                mergeEventService.addMergeIgmEvent(processConfig.getProcess(), processConfig.getBusinessProcess(), dateTime, tso, IgmStatus.AVAILABLE, null, null, null, null);
+            });
 
-                    for (ProcessConfig processConfig : processConfigs.subList(1, processConfigs.size())) {
-                        // import IGM into the network store
-                        UUID processConfigNetworkUuid = caseFetcherService.importCase(caseUuid);
-                        merge(processConfig, dateTime, date, tso, valid, processConfigNetworkUuid, caseUuid, businessProcess, null, null);
-                    }
+            // Mergings
+            List<Boolean> igmQuality = new ArrayList<>();
```

Comment:
Maybe an Optional instead of a List which, here, always contains one element ...

Reply:
No `AtomicBoolean` must be initialized with a value
This forces to use 2 `AtomicBoolean`